cognite_toolkit/_cdf_tk/prototypes/_packages/update_references.py

```diff
@@ -0,0 +1,338 @@
+                params[to_camel_case(param.name)] = [{"space": "str", "external_id": "str"}, "None"]
+                continue
+
+        elif cls.__name__ == "WorkflowDefinitionUpsert" or cls.__name__ == "WorkflowVersionUpsert":
+            if param.name == "workflow_definition":
+                params[to_camel_case(param.name)] = {
+                    "description": "str | None",
+                    "tasks": [
+                        {
+                            "externalId": "str",
+                            "type": "function | transformation | cdf",
+                            "parameters": {
+                                "function": {
+                                    "externalId": "str",
+                                }
+                            },
+                        }
+                    ],
+                }
+                continue
+
+        elif cls.__name__ == "GroupWrite":
+            if param.name == "capabilities":
+                params[to_camel_case(param.name)] = expand_acls()
+                continue
+
+        elif cls.__name__ == "DataModelApply":
+            if param.name == "views":
+                params[to_camel_case(param.name)] = {
+                    "*optional list[ViewId | ViewApply]": {
+                        "*one list of": [expand_parameters(cls, ViewId, True), expand_parameters(cls, ViewApply, True)]
+                    }
+                }
+                continue
+
+        if str(param.annotation) == "dict[str, str] | None":
+            params[to_camel_case(param.name)] = {
+                "*optional dict[str,str]": {"*key1": "value1", "*key2": "value2"},
+            }
+            continue
+
+        if param.name == "data_set_id":
+            params["dataSetExternalId"] = param.annotation.replace("int", "str")
+            continue
+
+        params[to_camel_case(param.name)] = f"{'*' if optional else ''}{unpack_annotations(str(param.annotation))}"
+
+    return params
+
+
+def generate(target_path: Path) -> None:
+    for loader in [
+        loader for loader in ResourceLoader.__subclasses__() if loader.__name__ not in ["ResourceContainerLoader"]
+    ]:
+        write_cls = loader.resource_write_cls  # type: ignore
+
+        ref = expand_parameters(loader, write_cls)
+
+        final_yaml = comment_optional(yaml.dump(ref, sort_keys=False))
+
+        # special case for groups: want caps at the end
+        if "capabilities" in ref:
+            caps_yaml = comment_optional(yaml.dump({"capabilities": ref.pop("capabilities")}, sort_keys=False))
+            main_yaml = comment_optional(yaml.dump(ref, sort_keys=False))
+            final_yaml = f"{main_yaml}{caps_yaml}"
+
+        folder = target_path / loader.folder_name
+        file_name = folder / Path(f'reference.{write_cls.__name__.replace("Write","").replace("Apply","")}.yaml')
+        Path.mkdir(folder, exist_ok=True)
+        Path.write_text(file_name, final_yaml)
```

Comment:
`loader_cls` to make it clear that this is a a class, and we already have `kind` for this exact purpose
```suggestion
    for loader_cls in [
        loader for loader in ResourceLoader.__subclasses__() if loader.__name__ not in ["ResourceContainerLoader"]
    ]:
        write_cls = loader_cls.resource_write_cls  # type: ignore

        ref = expand_parameters(loader_cls, write_cls)

        final_yaml = comment_optional(yaml.dump(ref, sort_keys=False))

        # special case for groups: want caps at the end
        if "capabilities" in ref:
            caps_yaml = comment_optional(yaml.dump({"capabilities": ref.pop("capabilities")}, sort_keys=False))
            main_yaml = comment_optional(yaml.dump(ref, sort_keys=False))
            final_yaml = f"{main_yaml}{caps_yaml}"

        folder = target_path / loader_cls.folder_name
        file_name = folder / Path(f'reference.{loader_cls.kind}.yaml')
        Path.mkdir(folder, exist_ok=True)
        Path.write_text(file_name, final_yaml)

        print(f"Wrote {file_name}")
```